File: examples/local_script.py
from blackboxd import OpenAI, configure, trace_llm, trace_span


configure(
    storage=".blackboxd/local-script.jsonl",
    environment="local",
    app_version="0.1.0",
    default_tags=["batch-review"],
)

client = OpenAI()


@trace_llm(tags=["receipt-review"])
def review_receipt(receipt_text: str) -> str:
    with trace_span("classify", metadata={"document_type": "receipt"}):
        response = client.responses.create(
            model="gpt-4.1-mini",
            input=[
                {
                    "role": "user",
                    "content": [{"type": "input_text", "text": receipt_text}],
                }
            ],
        )
    return response.output_text


if __name__ == "__main__":
    print(review_receipt("Store: Coffee Lab, Amount: 12.40 USD, Please summarize."))

File: typescript/examples/local-script.ts
import { configure, JSONLStorage, OpenAIWrapper, trace, traceSpan } from "../src/index.ts";

configure({
  storage: new JSONLStorage(".blackboxd/ts-local-script.jsonl"),
  environment: "local",
  appVersion: "0.1.0",
  defaultTags: ["typescript"],
});

const fakeClient = {
  responses: {
    async create(input: Record<string, unknown>) {
      return {
        id: "resp_123",
        model: input.model,
        output_text: "Approved",
        usage: { input_tokens: 12, output_tokens: 4 },
      };
    },
  },
};

const client = new OpenAIWrapper(fakeClient);

const reviewReceipt = trace("reviewReceipt", { tags: ["batch-review"] })(async (text: string) => {
  return traceSpan("classify", async () => {
    const response = (await client.responses.create({
      model: "gpt-4.1-mini",
      input: `Review this receipt: ${text}`,
    })) as { output_text: string };

    return response.output_text;
  });
});

console.log(await reviewReceipt("Store: Coffee Lab, Amount: 12.40 USD"));
